javalib/src/main/scala/java/time/Duration.scala

```diff
@@ -0,0 +1,234 @@
+package java.time
+
+import scala.collection.JavaConverters._
+
+import java.time.temporal._
+
+final class Duration private(seconds: Long, nanos: Int)
+    extends TemporalAmount with Comparable[Duration]
+    with java.io.Serializable with Preconditions {
+
+  require(nanos >= 0 && nanos <= 999999999, "nanos must be >= 0 and <= 999999999")
```

Comment:
Could you be more specific about this? Note that the package-private trait Preconditions overrides require here to throw a DateTimeException here. Would it be okay to define a custom method `requireDT` in the package object instead?